spec/rubocop/cop/lint/non_deterministic_require_order_spec.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Lint::NonDeterministicRequireOrder do
+  subject(:cop) { described_class.new(config) }
+
+  let(:config) { RuboCop::Config.new }
+
+  context 'when requiring files' do
+    context 'with unsorted index' do
+      it 'registers an offsense' do
+        expect_offense(<<~RUBY)
+          Dir["./lib/**/*.rb"].each do |file|
+          ^^^^^^^^^^^^^^^^^^^^^^^^^ Sort files before requiring them.
+            require file
+          end
+        RUBY
+      end
+
+      it 'registers an offsense with extra logic' do
+        expect_offense(<<~RUBY)
+          Dir["./lib/**/*.rb"].each do |file|
+          ^^^^^^^^^^^^^^^^^^^^^^^^^ Sort files before requiring them.
+            if file.starts_with('_')
+              puts "Not required."
+            else
+              require file
+            end
+          end
+        RUBY
+      end
+    end
+
+    context 'with unsorted glob' do
+      it 'registers an offsense' do
+        expect_offense(<<~RUBY)
+          Dir.glob(Rails.root.join(__dir__, 'test', '*.rb'), File::FNM_DOTMATCH).each do |file|
```

Comment:
Regarding your question
> Should it test for more patterns?

I think it might make sense to support this case as well from the beginning:
```ruby
Dir.glob(Rails.root.join(__dir__, 'test', '*.rb'), File::FNM_DOTMATCH) do |file|
  require file
end
```
(that is `glob` taking a block without `each`)

What do you think? (Adding this should be rather easy with a `{(send …) (send …)}` alternative in `unsorted_dir_loop?`.)

Reply:
Yes! Good catch. We should definitely support that.

Reply:
Is this the best way to write the new pattern?
```
{ (send (const nil? :Dir) :glob ...) (send (send (const nil? :Dir) {:[] :glob} ...) :each) }
```

Reply:
(Only the `.glob` variant supports the direct block syntax)

Reply:
I ended up splitting it into two patterns to help with autocorrection.